### PR TITLE
Disabled nav button clicks will not fire click event

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#35

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks](https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks](https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-progress-indicator?_sp.spa=pr-disabled-nav-button-clicks](https://developer.blackbaud.com/skyux-progress-indicator?_sp.spa=pr-disabled-nav-button-clicks)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-disabled-nav-button-clicks)

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Looks good. Should we have a test for this?

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-AlexKingman we can discuss this but this one is extremely hard to test due to the use of css `pointer-events`. Unit test clicks will still call the click handler as the `click` and `triggerEventHandlers` call the event handlers directly and don't mimic real world browser use where the css will stop the click events from firing. I've tried pretty much everything I can think of to get tests that correctly fail prior to this change and pass after but have had no luck. Happy to hop on a call with you to discuss more and if there are any options if you'd like.

File: src/app/public/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts [hunk 1]
```diff
@@ -79,20 +81,24 @@ export class SkyProgressIndicatorNavButtonComponent implements AfterViewInit, On
...
     this._disabled = value;
     this.changeDetector.markForCheck();
   }
 
   public get disabled(): boolean {
     const buttonType = this.buttonType;
-    const activeIndex = this.lastProgressChange.activeIndex;
-    const isLastStep = (activeIndex === this.lastProgressChange.itemStatuses.length - 1);
+    const activeIndex = this.lastProgressChange ? this.lastProgressChange.activeIndex : 0;
+    const isLastStep = this.lastProgressChange ? (activeIndex === this.lastProgressChange.itemStatuses.length - 1) : false;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts:101]: These changes need unit tests, no?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
For the click event unit test (asked for by @Blackbaud-AlexKingman): could you add an Angular `(click)` handler to the `<sky-progress-indicator-nav-button>` component and detect if the click handler fires? Isn't that what the issue is trying to solve? 
```
<sky-progress-indicator-nav-button
  buttonType="finish"
  [disabled]="disabled"
  (click)="onClick()" <-- This should never get called.
>
```

Maybe `pointer-events` isn't the correct solution; the fix should also prevent explicit calls to the element's `.click()` method.
```
<sky-progress-indicator-nav-button
  class="my-button"
  [disabled]="disabled"
  (click)="onClick()"
>

// This shouldn't work either:
document.querySelector('.my-button').click();
```

Is this possible?

**Comment by Blackbaud-TrevorBurch** [comment]
Going to transition this to documentation changes

**Pull request closed without merging**